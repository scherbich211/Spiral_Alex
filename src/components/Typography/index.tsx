/* eslint-disable react/no-unused-prop-types */
import * as React from 'react';
import {LayoutChangeEvent, StyleSheet, TextStyle} from 'react-native';
import {Text, useTheme} from 'react-native-paper';

type TypographyElement = 'h1' | 'h2' | 'h3' | 'body' | 'button' | 'caption';
type FontWeight = 'normal' | 'bold' | '100' | '200' | '300' | '400' | '500' | '600' | '700' | '800' | '900';

const getFontSize = (element?: TypographyElement): number => {
	switch (element) {
		case 'h1':
			return 24;
		case 'h2':
			return 24;
		case 'h3':
			return 20;
		case 'body':
			return 16;
		case 'button':
			return 14;
		case 'caption':
			return 11;
		default:
			return 14;
	}
};

const getFontFamily = (element?: TypographyElement): string => {
	switch (element) {
		case 'h1':
			return 'Roboto-Regular';
		case 'h2':
			return 'Roboto-Regular';
		case 'h3':
			return 'Roboto-Regular';
		case 'body':
			return 'RobotoCondensed-Regular';
		case 'button':
			return 'Roboto-Regular';
		case 'caption':
			return 'RobotoCondensed-Regular';
		default:
			return 'RobotoCondensed-Regular';
	}
};

const getFontWeight = (element?: TypographyElement): FontWeight => {
	switch (element) {
		case 'h1':
			return 'bold';
		case 'h2':
			return 'bold';
		case 'h3':
			return 'bold';
		case 'body':
			return 'normal';
		case 'button':
			return '500';
		case 'caption':
			return 'normal';
		default:
			return 'normal';
	}
};

interface Props {
	element?: TypographyElement;
	fontSize?: number;
	fontWeight?: FontWeight;
	letterSpacing?: number;
	fontFamily?: string;
	children: React.ReactNode;
	style?: TextStyle;
	onLayout?: (event: LayoutChangeEvent) => void;
	color?: TextStyle['color'];
}

const Typography: React.FunctionComponent<Props> = props => {
	const theme = useTheme();
	const {children, onLayout} = props;
	const styles = useStyles({props: {...props}, theme});
	return (
		<Text style={StyleSheet.flatten([styles.main, props.style])} onLayout={onLayout}>
			{children}
		</Text>
	);
};

const useStyles = ({props, theme}: {props: Props; theme: ReactNativePaper.Theme}) =>
	StyleSheet.create({
		main: {
			fontSize: props.fontSize ? props.fontSize : getFontSize(props.element),
			fontFamily: props.fontFamily ? props.fontFamily : getFontFamily(props.element),
			fontWeight: props.fontWeight ? props.fontWeight : getFontWeight(props.element),
			color: props.color ? props.color : theme.colors.text,
		},
	});

export default Typography;
