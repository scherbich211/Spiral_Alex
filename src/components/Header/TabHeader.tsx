import React from 'react';
import {Platform, StyleSheet, TextStyle, View} from 'react-native';
import {Appbar, useTheme} from 'react-native-paper';
import Typography from '../Typography';
import {SIZES} from '../../theme';

interface Props {
	headerText?: string;
	children?: React.ReactNode;
	beforeText?: React.ReactNode;
	headerStyle?: TextStyle;
}

const TabHeader: React.FC<Props> = props => {
	const theme = useTheme();
	return (
		<Appbar.Header
			theme={{
				colors: {
					primary: theme?.colors.primary,
				},
			}}
			style={styles.headerView}>
			{props.beforeText ? (
				<View style={styles.headerBeforeView}>
					{props.beforeText}
					<Typography element="h3" style={{...styles.headerText, ...props.headerStyle}}>
						{props.headerText}
					</Typography>
				</View>
			) : (
				<View>
					<Typography element="h3" style={styles.headerText}>
						{props.headerText}
					</Typography>
				</View>
			)}
			{props.children}
		</Appbar.Header>
	);
};

const styles = StyleSheet.create({
	headerView: {
		alignItems: 'center',
		elevation: 2,
		zIndex: 2,
	},
	headerText: {
		fontWeight: '700',
		marginLeft: SIZES.width * 0.3,
		fontSize: Platform.OS === 'ios' ? 18 : 17,
		letterSpacing: Platform.OS === 'ios' ? 1 : 0.8,
		color: 'white',
		// textTransform: 'uppercase',
	},
	headerBeforeView: {
		flexDirection: 'row',
		alignItems: 'center',
		flex: 1,
	},
});

export default TabHeader;
