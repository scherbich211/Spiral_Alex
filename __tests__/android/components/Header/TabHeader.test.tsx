import React from 'react';
import {waitFor, render, cleanup} from '@testing-library/react-native';
import TabHeader from '../../../../src/components/Header/TabHeader';
import { Text } from 'react-native';
import Typography from '../../../../src/components/Typography';

jest.mock('react-native', function () {
	const reactNative = jest.requireActual('react-native');
	jest.spyOn(reactNative.Platform, 'select').mockImplementation(() => 'android');
	reactNative.Platform.OS = 'android';
	return reactNative;
});

describe('Tab Header', () => {
	afterEach(cleanup);

	it('Render', async () => {
		const Header = await waitFor(() =>
			render(<TabHeader />),
		);
		expect(Header).toBeDefined();
		expect(Header).toMatchSnapshot();
		Header.unmount();
	});

	it('Render & headerText', async () => {
		const Header = await waitFor(() =>
			render(<TabHeader headerText='Home'/>),

		);
		const textShow = await Header.getByText('Home')
		expect(textShow).toBeDefined();
		expect(Header).toMatchSnapshot();
		Header.unmount();
	});

	it('Render & children', async () => {
		const Header = await waitFor(() =>
			render(<TabHeader><Typography>Test</Typography></TabHeader>),

		);
		const children = await Header.getByText('Test')
		expect(children).toBeDefined();
		expect(Header).toMatchSnapshot();
		Header.unmount();
	});

	it('Render & beforeText', async () => {
		const Header = await waitFor(() =>
			render(<TabHeader beforeText={<Typography testID='Test'>Test</Typography>}></TabHeader>),

		);
		const beforeText = await Header.queryByTestId('Test')
		expect(beforeText).toBeDefined();
		expect(Header).toMatchSnapshot();
		Header.unmount();
	});

	it('Render & headerText & headerStyle & beforeText', async () => {
		const Header = await waitFor(() =>
			render(
			<TabHeader 
				headerStyle={{backgroundColor: 'red'}} 
				headerText='Home' 
				beforeText={<Typography testID='Test'>Test</Typography>}
			>
			</TabHeader>),
		);
		const textShow = await Header.getByText('Home')
		expect(textShow).toBeDefined();
		const beforeText = await Header.queryByTestId('Test')
		expect(beforeText).toBeDefined();
		expect(Header.getByText('Home')).toHaveStyle({backgroundColor: 'red'})
		expect(Header).toMatchSnapshot();
		Header.unmount();
	});
});
