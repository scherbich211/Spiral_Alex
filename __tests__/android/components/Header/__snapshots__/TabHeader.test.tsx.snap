// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab Header Render & beforeText 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#6200ee",
        "elevation": 2,
        "zIndex": 2,
      },
      Object {
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.75,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 1.5,
      },
    ]
  }
>
  <View
    collapsable={false}
    nativeID="animatedComponent"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#6200ee",
        "elevation": 0,
        "flexDirection": "row",
        "height": 56,
        "marginTop": "android",
        "paddingHorizontal": 4,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Text
        testID="Test"
      >
        Test
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
            },
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "white",
              "fontFamily": "Roboto-Regular",
              "fontSize": 17,
              "fontWeight": "700",
              "letterSpacing": 0.8,
              "marginLeft": 187.5,
            },
          ]
        }
        testID="undefined"
      />
    </View>
  </View>
</View>
`;

exports[`Tab Header Render & children 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#6200ee",
        "elevation": 2,
        "zIndex": 2,
      },
      Object {
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.75,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 1.5,
      },
    ]
  }
>
  <View
    collapsable={false}
    nativeID="animatedComponent"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#6200ee",
        "elevation": 0,
        "flexDirection": "row",
        "height": 56,
        "marginTop": "android",
        "paddingHorizontal": 4,
      }
    }
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
            },
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "white",
              "fontFamily": "Roboto-Regular",
              "fontSize": 17,
              "fontWeight": "700",
              "letterSpacing": 0.8,
              "marginLeft": 187.5,
            },
          ]
        }
        testID="undefined"
      />
    </View>
    <Text>
      Test
    </Text>
  </View>
</View>
`;

exports[`Tab Header Render & headerText & headerStyle & beforeText 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#6200ee",
        "elevation": 2,
        "zIndex": 2,
      },
      Object {
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.75,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 1.5,
      },
    ]
  }
>
  <View
    collapsable={false}
    nativeID="animatedComponent"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#6200ee",
        "elevation": 0,
        "flexDirection": "row",
        "height": 56,
        "marginTop": "android",
        "paddingHorizontal": 4,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Text
        testID="Test"
      >
        Test
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
            },
            Object {
              "textAlign": "left",
            },
            Object {
              "backgroundColor": "red",
              "color": "white",
              "fontFamily": "Roboto-Regular",
              "fontSize": 17,
              "fontWeight": "700",
              "letterSpacing": 0.8,
              "marginLeft": 225,
            },
          ]
        }
        testID="undefined"
      >
        Home
      </Text>
    </View>
  </View>
</View>
`;

exports[`Tab Header Render & headerText 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#6200ee",
        "elevation": 2,
        "zIndex": 2,
      },
      Object {
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.75,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 1.5,
      },
    ]
  }
>
  <View
    collapsable={false}
    nativeID="animatedComponent"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#6200ee",
        "elevation": 0,
        "flexDirection": "row",
        "height": 56,
        "marginTop": "android",
        "paddingHorizontal": 4,
      }
    }
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
            },
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "white",
              "fontFamily": "Roboto-Regular",
              "fontSize": 17,
              "fontWeight": "700",
              "letterSpacing": 0.8,
              "marginLeft": 225,
            },
          ]
        }
        testID="undefined"
      >
        Home
      </Text>
    </View>
  </View>
</View>
`;

exports[`Tab Header Render 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#6200ee",
        "elevation": 2,
        "zIndex": 2,
      },
      Object {
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.75,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 1.5,
      },
    ]
  }
>
  <View
    collapsable={false}
    nativeID="animatedComponent"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#6200ee",
        "elevation": 0,
        "flexDirection": "row",
        "height": 56,
        "marginTop": "android",
        "paddingHorizontal": 4,
      }
    }
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
            },
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "white",
              "fontFamily": "Roboto-Regular",
              "fontSize": 17,
              "fontWeight": "700",
              "letterSpacing": 0.8,
              "marginLeft": 187.5,
            },
          ]
        }
        testID="undefined"
      />
    </View>
  </View>
</View>
`;
